pkg/ctl/update/nodegroup_test.go

```diff
@@ -0,0 +1,105 @@
+package update
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/ctl/ctltest"
+)
+
+var _ = Describe("update nodegroup", func() {
+	It("returns error if config file is not set", func() {
+		cmd := newMockCmd("nodegroup")
+		_, err := cmd.execute()
+		Expect(err).To(HaveOccurred())
+		Expect(err).To(MatchError(ContainSubstring("--config-file must be set")))
+	})
+
+	It("returns error if nodegroup is not set in config", func() {
+		cfg := &api.ClusterConfig{
+			TypeMeta: api.ClusterConfigTypeMeta(),
+			Metadata: &api.ClusterMeta{
+				Name:   "cluster-1",
+				Region: "us-west-2",
+			},
+		}
+		config := ctltest.CreateConfigFile(cfg)
+		cmd := newMockCmd("nodegroup", "--config-file", config)
+		_, err := cmd.execute()
+		Expect(err).To(HaveOccurred())
+		Expect(err).To(MatchError(ContainSubstring("managedNodeGroups field must be set")))
+	})
+
+	It("returns error if cfg contains unsupported field in NodeGroupBase fields", func() {
+		cfg := &api.ClusterConfig{
+			TypeMeta: api.ClusterConfigTypeMeta(),
+			Metadata: &api.ClusterMeta{
+				Name:   "cluster-1",
+				Region: "us-west-2",
+			},
+			ManagedNodeGroups: []*api.ManagedNodeGroup{
+				{
+					NodeGroupBase: &api.NodeGroupBase{
+						Name:           "ngName",
+						MaxPodsPerNode: 5,
+					},
+				},
+			},
+		}
+		config := ctltest.CreateConfigFile(cfg)
+		cmd := newMockCmd("nodegroup", "--config-file", config)
+		_, err := cmd.execute()
+		Expect(err).To(HaveOccurred())
+		Expect(err).To(MatchError(ContainSubstring("unsupported field: MaxPodsPerNode cannot be used with `eksctl update nodegroup`")))
+	})
+
+	It("returns error if cfg contains unsupported field in NodeGroup fields", func() {
```

Comment:
I may have missed something so this question may be irrelevant, but are we validating other fields for any other reason apart from clarity? Presumably we are only reading the `updateConfig` field for this command?

Reply:
it's just for clarity :)

Reply:
cool cool